xlgui/widgets/playlist_columns.py

```diff
@@ -168,44 +168,51 @@ def __repr__(self):
         )
 
 
-class CustomizedCellRendererText(Gtk.CellRendererText):
-    '''
-        The default GTK cell renderer triggers editing when any part of the column is
-        clicked. This is undesirable, as tracks can never be played if we allow columns
-        to be edited!
-
-        Instead, this only triggers editing if the mouse clicks the text of the
-        column.
-    '''
-
-    def do_start_editing(self, event, widget, path, background_area, cell_area, flags):
-        if event:
-            has_coords, x, y = event.get_coords()
-            if has_coords:
-                # The 'aligned area' seems to correspond to where the text actually
-                # is rendered. If the coordinates don't overlap that, then we don't
-                # allow the edit to occur. Conveniently, this should deal with RTL
-                # issues as well (not tested)
-                aa = self.get_aligned_area(widget, flags, cell_area)
-
-                if x < aa.x or x > aa.x + aa.width:
-                    return None
-
-        return Gtk.CellRendererText.do_start_editing(
-            self, event, widget, path, background_area, cell_area, flags
-        )
-
-
 class EditableColumn(Column):
-    cellproperties = {'editable': True}
-    renderer = CustomizedCellRendererText
+    renderer = Gtk.CellRendererText
 
     def __init__(self, *args):
         Column.__init__(self, *args)
         self.cellrenderer.connect('edited', self.on_edited)
         self.cellrenderer.connect('editing-started', self.on_editing_started)
+        self.cellrenderer.connect('editing-canceled', self._reset_editable)
+
+    def could_edit(self, treeview, path, event):
+        # Returns True if an edit could take place
+
+        # The 'aligned area' seems to correspond to where the text actually
+        # is rendered. If the coordinates don't overlap that, then we don't
+        # allow the edit to occur. Conveniently, this should deal with RTL
+        # issues as well (not tested)
+        cell_area = treeview.get_cell_area(path, self)
+
+        # Ensure the renderer text attribute is updated before finding
+        # it's width
+        model = treeview.get_model()
+        iter = model.get_iter(path)
+        self.data_func(self, self.cellrenderer, model, iter, None)
+
+        aa = self.cellrenderer.get_aligned_area(treeview, 0, cell_area)
+        x = event.x
+
+        if x >= aa.x and x <= aa.x + aa.width:
+            return True
+
+        return False
```

Comment:
This makes me itch to change the last few lines to to `return ... <= ... <= ...`, but whichever you think is more readable.